app/Http/Controllers/ArticleController.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Helper;
+use App\Http\Requests\ArticleStoreRequest;
+use App\Http\Requests\ArticleUpdateRequest;
+use App\Models\Article;
+use App\Models\Category;
+use App\Models\Tag;
+use App\Services\ArticleService;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Storage;
+
+class ArticleController extends Controller
+{
+    public function index()
+    {
+        return view('articles.index')
+            ->with('articles', Article::with('category')->orderBy('id', 'desc')->paginate(10));
+    }
+
+    public function create()
+    {
+        return view('articles.create')
+            ->with('categories', Category::all());
+    }
+
+    public function store(ArticleStoreRequest $request)
+    {
+        if ($request->has('image')) {
+            $request->file('image')->storeAs(
+                'uploads',
+                $fileName = Helper::generateRandomString() . '.' . $request->file('image')->getClientOriginalExtension(),
+                'public'
+            );
+        }
+
+        $article = Article::create(array_merge(
+            $request->validated(),
+            ['image' => $fileName ?? null, 'user_id' => auth()->user()->id]
+        ));
+
+        $tags = explode(',', $request->tags);
+        foreach ($tags as $tag) {
+            $article->tags()->attach(Tag::select('id')->where('name', $tag)->get());
```

Comment:
Here it would do select for every tag witch could be very bad for server usage. And what if tag isn't in db created? Then it won't be selected and attached right? So use maybe `updateOrCreate` and then attach tag to article. Something like that maybe?
```php
foreach ($tags as $tag) {
    $tagId = Tag::updateOrCreate(['name' => $tag]);
    $article->tags()->attach($tagId);
}
```